bounce.py

```
@@ -23,21 +22,82 @@
+
 
 @app.route('/')
 def bounce():
+    client_ip = get_client_ip()
+    if is_rate_limited(client_ip):
+        logging.warning(f"Rate limit exceeded for IP: {client_ip}")
+        return "Rate limit exceeded. Please try again later.", 429
+
     # Extract the domain from the Host header
     domain = request.headers.get('Host', 'unknown')
     logging.info(f"Received request for domain: {domain}")
```

Ruff

Logging statement uses f-string

(G004)